### PR TITLE
Initial support for test component

### PR DESCRIPTION
**Pull request by kennykerr**
This update adds support for the TestWinRT component:

https://github.com/microsoft/testwinrt

This WinRT component is written in C++/WinRT and provides more comprehensive test coverage than is possible by simply exercising the Windows API. 

This initial commit mainly provides tweaks to code generation to actually support parsing the test component's winmd file. Exhaustive test coverage of the test component's many test methods is coming up next!

### CODE REVIEW

File: crates/gen/src/types/delegate.rs [hunk 1]
```diff
@@ -58,6 +58,32 @@ impl Delegate {
             .map(|param| param.to_invoke_arg_tokens());
         let debug = debug::default_debug_tokens(&self.name);
 
+        let invoke_upcall = if let Some(return_type) = &self.method.return_type {
```
Comment (kennykerr) [inline crates/gen/src/types/delegate.rs:61]: This is just roughed in to allow the resulting Rust code to compile - it doesn't actually work yet. That's up next and I'm sure will change.

File: crates/gen/src/types/method.rs [hunk 2]
```diff
@@ -67,7 +67,7 @@ impl Method {
         let return_type = if blob.read_expected(0x01) {
             None
         } else {
-            let name = "__result".to_owned();
+            let name = "result__".to_owned();
```
Comment (kennykerr) [inline crates/gen/src/types/method.rs:70]: Switched from leading to trailing underscores for internal generated names to allow unused variables to quickly be detected.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -15,6 +15,12 @@ jobs:
     steps:
     - uses: actions/checkout@v2
 
+    - name: cargo winrt
+      run: cargo install --git https://github.com/microsoft/winrt-rs cargo-winrt
```
Comment (rylev) [inline .github/workflows/build.yml:19]: Let's not forget to change this with we release a new version

File: src/abi_transferable.rs [hunk 4]
```diff
@@ -23,6 +23,14 @@ pub unsafe trait AbiTransferable {
         // correctly implement this trait
         unsafe { std::mem::transmute_copy(&abi) }
     }
+
+    fn slice_from_abi<'a>(abi: *const Self::Abi, len: usize) -> &'a [Self] {
```
Comment (rylev) [inline src/abi_transferable.rs:27]: I think this function should be unsafe since it's up to the caller to ensure the pointer is valid for `'a`

Comment (kennykerr) [reply src/abi_transferable.rs:27]: Agreed.

**Pull request merged**